### ISSUE TITLE
required executables

### ISSUE THREAD

**Issue by nsheff** [epigen/looper#23]
It would be nice if a pipeline could specify what executables it needs.

This could happen in the `pipeline_interface` -- a  new attribute called `required_executables`, and perhaps a second called `required_config_executables`, could be read by looper, which could test these executables to make sure they are in the PATH.

**Comment by afrendeiro** [comment]
There was some groundwork for this in the `CommandChecker` class. I guess it could be adapted/integrated in that.

**Comment by vreuter** [comment]
Feel free to de-escalate if you wish @nsheff but this seems valuable...

**Comment by vreuter** [comment]
Could we infer this? Something like:
1 -- determine pipelines that should be run and from which pipelines location
2 -- grab config file for each such pipeline (from the relevant location)
3 -- parse the `tools` and `resources` sections of the config
4 -- determine which pipeline(s)--if any--are missing necessary environment variable/path settings and provide such feedback, either proceeding with the pipelines that are OK, or failing altogether

**Comment by nsheff** [comment]
This could lead to a new command like `looper executable` that would test pipelines for if they will run or not...

**Comment by vreuter** [comment]
I like that idea. What about `execs`? Or making `check` be `flags` and letting this concept be `check`?

**Comment by nsheff** [comment]
I like both suggestions.

**Comment by nsheff** [comment]
I don't think it would be possible to infer this. I don't really want to impose structure on the config file here, and it's not just config settings that would determine executables.

I think the only way to really do it well is for the pipeline author to specify a list, and the best place for that is in the pipeline interface.

**Comment by vreuter** [comment]
I agree; should we implement the pipeline interface specification way for this version?

**Comment by nsheff** [comment]
if it's not too much work. otherwise we can wait for 0.8
